src/protocols/secure_channel/CASESession.cpp

```diff
@@ -187,18 +187,18 @@ CHIP_ERROR CASESession::FromCachable(const CASESessionCachable & cachableSession
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR CASESession::Init(uint16_t localSessionId, SessionEstablishmentDelegate * delegate)
+CHIP_ERROR CASESession::Init(SessionHolder & secureSessionHolder, SessionEstablishmentDelegate * delegate)
```

Comment:
Should this take `SessionHolder &&`?  Conceptually the consumer is handing over ownership of the session to this class, right?

Reply:
SessionHolder is more like a shared pointer, and the copy and move constructors enforce this.  Everybody who has a reference is an owner, and the code respects that.  The session object can also reach into all holders and release their session reference, so retaining a reference can't break things.

However, in this case, I can't think of a reason a caller would need to retain a reference.  I'll give this a shot.

Reply:
Why even have this model of allocating the session in one place, and then transferring ownership? Why wouldn't we just have `CASESession` allocate the session, instead of having someone else do it?

That would mean exchange allocation would also have to shift into this (which I think is the right pattern, since CASESession effectively manages the state machine and message processing for the CASE protocol, so it should be the one to allocate the EC).

Reply:
That is even better.  Will do.

Reply:
Done.